### PR TITLE
chore: improve type checks in the stack attributes.

### PR DESCRIPTION
**Pull request by i4ki**

### CODE REVIEW

File: hcl/hcl.go [hunk 1]
```diff
@@ -639,28 +639,43 @@ func assignSet(name string, target *[]string, val cty.Value) error {
...
 	for iterator.Next() {
+		index++
 		_, elem := iterator.Element()
 
 		logger.Trace().Msg("Check element is of correct type.")
 
 		if elem.Type() != cty.String {
-			errs.Append(errors.E("field %q is a set(string) but contains %q",
-				name, elem.Type().FriendlyName()))
+			errs.Append(errors.E("field %q must be a set(string) but element %d "+
```
Comment (katcipis) [inline hcl/hcl.go:666]:
```suggestion
			errs.Append(errors.E("field %q must be a string but element %d "+
```

**Review by katcipis — COMMENTED** [review]
The error message references set(string) but I think it is supposed to be just string

File: hcl/hcl.go (hunk 1 above)
Comment (i4ki) [reply hcl/hcl.go:666]: this generates the error: `field "wants" must be a set(string) but element 1 has type "number"`.
I'm using the terraform type declaration here.. not sure if ok, but if then it says the `wants` must be a `set` (unique values) of `string` values.

**Comment by i4ki** [comment]
@soerenmartius can I have a quick review here?

**Review by katcipis — APPROVED** [review]

**Pull request merged**